crates/partition-store/src/outbox_table/mod.rs

```diff
@@ -82,14 +82,35 @@ fn get_outbox_message<S: StorageAccess>(
 fn truncate_outbox<S: StorageAccess>(
     storage: &mut S,
     partition_id: PartitionId,
-    seq_to_truncate: Range<u64>,
+    range: RangeInclusive<u64>,
 ) {
-    let mut key = OutboxKey::default().partition_id(partition_id);
-    let k = &mut key;
+    let _x = RocksDbPerfGuard::new("truncate-outbox");
+
+    let mut keys = vec![];
+    {
+        let start_key = OutboxKey::default()
+            .partition_id(partition_id)
+            .message_index(range.start().clone());
+
+        let end_key = OutboxKey::default()
+            .partition_id(partition_id)
+            .message_index(range.end().clone());
+
+        let mut iter = storage.iterator_from(TableScan::KeyRangeInclusiveInSinglePartition(
+            partition_id,
+            start_key,
+            end_key,
+        ));
+
+        iter.seek_to_last();
+        while iter.valid() {
+            keys.push(iter.key().unwrap().to_vec());
+            iter.prev();
+        }
+    }
```

Comment:
I'd rather not merge this approach at all but curious if someone has a better way to implement this if we accept that scan-and-delete is actually an acceptable strategy here.

Reply:
Updated to an explicit counted loop.